skbio/alignment/_alignment.py

```diff
@@ -72,13 +72,16 @@ def __init__(self, seqs):
         self._data = seqs
         self._id_to_index = {}
         for i, seq in enumerate(self._data):
-            id = seq.id
-            if id in self:
+            if 'id' not in seq.metadata:
```

Comment:
Those tests are for pairwise alignment, where the output is an `Alignment` object with two sequences. If an input sequence doesn't have an id associated with it, the pairwise alignment code assigns an id (ascending integers starting from 0).

The changes proposed here (raising an error if a sequence doesn't have an id) shouldn't cause those tests to fail, so they should need to be changed/removed.